pkg/odo/cli/component/deploy.go

```diff
@@ -64,7 +67,29 @@ func (do *DeployOptions) Complete(name string, cmd *cobra.Command, args []string
 
 // Validate validates the push parameters
 func (do *DeployOptions) Validate() (err error) {
-	// TODO: Validate the value of tag and any user parameteres.
+
+	// Validate the --tag
+	if do.tag == "" {
+		return errors.New("odo deploy requires a tag, in the format <registry>/namespace>/<image>")
+	}
```

Comment:
Yeah I left a comment [here](https://github.ibm.com/hybrid-devops/appsody-internal/issues/18#issuecomment-21197251) mentioning that. It can moved to suit the Kubernetes vs OCP tracks when they're implemented.